typing/ctype.ml

```diff
@@ -1341,9 +1339,28 @@ let instance_constructor existential_treatment cstr =
             link_type tv to_unify;
             tv
     in
+    let copy_argument ty_arg_gen =
+      (* On patterns (Foo p) where p has an inline record type
+           | Foo of { x : ...; y : ... }
+         we do not want inline record type to escape the current clause,
+         so we lower its scope. *)
+      let ty_arg = copy copy_scope ty_arg_gen in
+      let is_inline_record_type ty =
+        match get_desc ty with
+        | Tconstr(p, _, _) when Path.is_constructor_typath p -> true
+        | _ -> false
+      in
+      let lower_scope ty =
```

Comment:
What do you mean by `lower_scope` ?
The scope is only supposed to be raised.

Note also that the scope of a type node is the scope of equations involving it, it is not the scope of its contents.
If you want to prevent the contents to leak, you should rather add an extra type parameter to the record definition, and put a a fresh existential there.

Reply:
Thinking a bit more, it should be enough to add an extra type parameter to inline records, and list it as existential in the constructor description of the enclosing variant type. Then pattern matching would automatically generate a fresh existential type for the argument, without any code to add.